ui/src/components/Equipment/Harvester.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { useState } from 'react';
+import { Card, Button, Badge } from 'react-bootstrap';
+import { contextMenu, Menu, Item, Separator } from 'react-contexify';
+
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import { MdContentCopy } from 'react-icons/md';
+import { FcRefresh, FcUpload, FcCollect } from 'react-icons/fc';
+
+import ImageViewer from '../ImageViewer/ImageViewer.jsx';
+
+import styles from './equipment.module.css';
+
+const sampleStateBackground = (key) => {
+  switch (key) {
+    case 'ready_to_execute': {
+      return 'success';
+    }
+    case 'harvested': {
+      return 'info';
+    }
+    case 'needs_repositionning': {
+      return 'warning';
+    }
+    case 'failed': {
+      return 'danger';
+    }
+    default: {
+      return 'info';
+    }
+  }
+};
+
+export default function Harvester(props) {
+  const [copied, setCopied] = useState(false);
+
+  function showContextMenu(event, id) {
+    let position = {
+      x: event.clientX,
+      y: event.clientY,
+    };
+    if (props.inPopover) {
+      position = {
+        x: event.offsetX,
+        y: event.offsetY,
+      };
+    }
+
+    contextMenu.show({
+      id,
+      event,
+      position,
+    });
+  }
+
+  const harvestCrystal = (UUID) => {
+    props.harvestCrystal(UUID);
+  };
+
+  const harvestAndLoadCrystal = (UUID) => {
+    props.harvestAndLoadCrystal(UUID);
+  };
+
+  const onCopy = () => {
+    setCopied(true);
+    setTimeout(() => setCopied(false), 1000);
+  };
+
+  const renderCrystalMenu = (key) => (
+    <Menu id={key}>
+      <Item onClick={() => harvestCrystal(key)}>
+        <span>
+          harvest Crystal <FcCollect />
+        </span>
+      </Item>
+      <Separator />
+      <Item onClick={() => harvestAndLoadCrystal(key)}>
+        <span>
+          <FcCollect /> harvest & Load Sample <FcUpload />
+        </span>
+      </Item>
+    </Menu>
+  );
+
+  const crystalUUID = props.contents.harvester_crystal_list;
```

Comment:
```suggestion
  const crystalUuid = props.contents.harvester_crystal_list;
```

`crystalUuid` is fine, no need for extra upper case letters (at least in my opinion)